site/app/controllers/grading/ImagesController.php

```diff
@@ -177,22 +182,37 @@ public function ajaxUploadImagesFiles() {
 
     /**
      * @Route("/courses/{_semester}/{_course}/flag_user_image", methods={"POST"})
-     * @AccessControl(role="INSTRUCTOR")
+     * @AccessControl(role="FULL_ACCESS_GRADER")
      */
-    public function flagUserImage() {
+    public function flagUserImage(): JsonResponse {
+        $user_id = $_POST['user_id'];
+
         if ($_POST['flag'] === 'true') {
             $new_state = 'flagged';
-            $success_msg = $_POST['user_id'] . '\'s image was successfully flagged.';
-            $fail_msg = 'Some error occurred flagging ' . $_POST['user_id'] . '\'s image.';
+            $icon_html = ImagesView::UNDO_ICON_HTML;
+            $href = "javascript:flagUserImage('$user_id', false)";
         }
         else {
             $new_state = 'preferred';
-            $success_msg = $_POST['user_id'] . '\'s image was successfully unflagged.';
-            $fail_msg = 'Some error occurred unflagging ' . $_POST['user_id'] . '\'s image.';
+            $icon_html = ImagesView::FLAG_ICON_HTML;
+            $href = "javascript:flagUserImage('$user_id', true)";
+        }
+
+        $result = $this->core->getQueries()->updateUserDisplayImageState($user_id, $new_state);
+        $user = $this->core->getQueries()->getSubmittyUser($user_id);
+
+        if ($result) {
+            $display_image = $user->getDisplayImage();
+
+            return JsonResponse::getSuccessResponse([
+                'first_last_username' => $user->getDisplayedFirstName() . ' ' . $user->getDisplayedLastName(),
+                'image_data' => $display_image->getImageBase64MaxDimension(ImagesView::IMG_MAX_DIMENSION),
+                'image_mime_type' => $display_image->getMimeType(),
```

Comment:
Add a conditional whether $display_image is null because when user doesn't have an image or when its flagged getDisplayImage returns null and its throwing PHP error (internally... I mean you dont see a normal error page with all the details as in other normal cases because of the JsonResponse type) when accessing methods on this null value for assigning values to $image_mime_type and $image_data assignments (L209-L210)

Reply:
Added `$display_image` null check.